libbeat/idxmgmt/lifecycle/es_client_handler.go

```diff
@@ -95,7 +95,7 @@ func NewESClientHandler(c ESClient, info beat.Info, cfg RawConfig) (*ESClientHan
 	// if the user has set both to different values, throw a warning, as overwrite operations will probably fail
 	if c.IsServerless() {
 		if cfg.TemplateName != "" && cfg.TemplateName != name {
-			logp.L().Warnf("policy name is %s but template name is %s; under serverless, non-default template and policy names should be the same. Updates & overwrites may not work.", name, cfg.TemplateName)
+			logp.L().Warnf("setup.dsl.data_stream_pattern is %s, but setup.template.name is %s; under serverless, non-default template and DSL pattern names should be the same. Updates & overwrites to this config may not work.", name, cfg.TemplateName)
```

Comment:
> Updates & overwrites to this config may not work.

What does this mean? Shouldn't they either work or not?

Reply:
Yah, good point. The way it works, the initial lifecycle policy happens as part of the datastream creation, so it doesn't matter if the name mismatches. Any attempted updates will fail, though.

Reply:
I assume it has to be deleted then? If so I would mention that specifically. I think this is much better than it was before, but it is best to tell the user exactly what to do to fix any error they are receiving.